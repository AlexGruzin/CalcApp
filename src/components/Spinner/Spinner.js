import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { CircleSpinner } from 'react-spinners-kit';

import './Spinner.styles.less';

const Spinner = ({ className, type, ...innerProps }) => (
  <div className={classnames('spinner', className, { [`spinner__${type}`]: type })}>
    <CircleSpinner loading {...innerProps} />
  </div>
);

Spinner.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
  frontColor: PropTypes.string,
  backColor: PropTypes.string,
  className: PropTypes.string,
  type: PropTypes.string,
};

Spinner.defaultProps = {
  size: 24,
  color: '#fff',
  frontColor: 'transparent',
  backColor: 'transparent',
  className: null,
  type: '',
};

export default Spinner;
