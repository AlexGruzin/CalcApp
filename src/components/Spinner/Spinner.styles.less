.spinner {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
}

.spinner > div {
  border-width: 5px;
}
