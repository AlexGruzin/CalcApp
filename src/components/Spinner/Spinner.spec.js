import React from 'react';
import { waitFor, render } from '@testing-library/react';

import Spinner from './Spinner';

describe('Spinner tests', () => {
  it('Check snapshots of Spinner are equal.', () => {
    const { asFragment } = render(<Spinner className="my-spinner" type="circle" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
