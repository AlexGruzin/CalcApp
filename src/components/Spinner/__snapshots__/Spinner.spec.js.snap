// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner tests Check snapshots of Spinner are equal. 1`] = `
<DocumentFragment>
  <div
    class="spinner my-spinner spinner__circle"
  >
    <div
      class="circle__Wrapper-sc-16bbsoy-0 dpSNSD"
      color="#fff"
      size="24"
    />
  </div>
</DocumentFragment>
`;
